src/sentry/consumers/__init__.py

```diff
@@ -237,8 +239,8 @@ def ingest_events_options() -> list[click.Option]:
         },
     },
     "generic-metrics-subscription-results": {
-        "topic": settings.KAFKA_GENERIC_METRICS_SUBSCRIPTIONS_RESULTS,
-        "default_topic": "generic-metrics-subscription-results",
+        "topic": Topic.GENERIC_METRICS_SUBSCRIPTIONS_RESULTS,
```

Comment:
Is there any specific reason this is the only topic being set via the `Topic` enum ?

Reply:
I'll give the same answer as above - two reasons:
- demonstration of how this works
- it is the only topic that has schema validation turned on. In order to keep that happening, we need to know both the default and override topic there so it must use the new system.